### PR TITLE
Update the 2.0.0 Upgrade Guide, misc. other docs

### PR DESCRIPTION
**Pull request by rileykarson**
* Semi-standardise on calling it the `2.0.0` Upgrade Guide > Version `2.0.0` Upgrade Guide.
 * Reflow, add some stuff, reformat.
* Rearrange the sidebar.
* Rename `google` provider reference -> [Google] Provider Configuration Reference

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: third_party/terraform/website/docs/version_2_upgrade.html.markdown [hunk 1]
```diff
@@ -60,11 +87,16 @@ Upgrade topics:
...
+without unexpected changes or deprecation notices.
 
--> Before upgrading to version 2.0.0, it is recommended to upgrade to the most recent version of the provider (1.19.1) and ensure that your environment successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html) without unexpected changes or deprecation notices.
-
-It is recommended to use [version constraints when configuring Terraform providers](https://www.terraform.io/docs/configuration/providers.html#provider-versions). If you are following that recommendation, update the version constraints in your Terraform configuration and run [`terraform init`](https://www.terraform.io/docs/commands/init.html) to download the new version.
+It is recommended to use [version constraints](https://www.terraform.io/docs/configuration/providers.html#provider-versions)
+when configuring Terraform providers. If you are following that recommendation,
+update the version constraints in your Terraform configuration and run
+[`terraform init`](https://www.terraform.io/docs/commands/init.html) to download
+the new version.
```
Comment (paddycarver) [inline third_party/terraform/website/docs/version_2_upgrade.html.markdown]: nit: we should probably update the example from 1.19.0 to 1.20.0.

File: third_party/terraform/website/docs/version_2_upgrade.html.markdown [hunk 2]
```diff
@@ -60,11 +87,16 @@ Upgrade topics:
...
+-> Before upgrading to version 2.0.0, it is recommended to upgrade to the most
+recent version of the provider (1.20.0) and ensure that your environment
+successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html)
+without unexpected changes or deprecation notices.
 
--> Before upgrading to version 2.0.0, it is recommended to upgrade to the most recent version of the provider (1.19.1) and ensure that your environment successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html) without unexpected changes or deprecation notices.
-
-It is recommended to use [version constraints when configuring Terraform providers](https://www.terraform.io/docs/configuration/providers.html#provider-versions). If you are following that recommendation, update the version constraints in your Terraform configuration and run [`terraform init`](https://www.terraform.io/docs/commands/init.html) to download the new version.
+It is recommended to use [version constraints](https://www.terraform.io/docs/configuration/providers.html#provider-versions)
+when configuring Terraform providers. If you are following that recommendation,
```
Comment (paddycarver) [inline third_party/terraform/website/docs/version_2_upgrade.html.markdown]: nit: we should probably tell people what to do if they _aren't_ following that recommendation (namely, `terraform init -upgrade`).

File: third_party/terraform/website/docs/version_2_upgrade.html.markdown (hunk 1 above)
Comment (rileykarson) [reply third_party/terraform/website/docs/version_2_upgrade.html.markdown]: I missed that one, thanks!

File: third_party/terraform/website/docs/version_2_upgrade.html.markdown (hunk 2 above)
Comment (rileykarson) [reply third_party/terraform/website/docs/version_2_upgrade.html.markdown]: 👍

**Pull request merged**